program-runtime/src/compute_budget.rs

```diff
@@ -24,9 +24,10 @@ impl ::solana_frozen_abi::abi_example::AbiExample for ComputeBudget {
 
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub struct ComputeBudget {
-    /// Number of compute units that an instruction is allowed.  Compute units
-    /// are consumed by program execution, resources they use, etc...
-    pub max_units: u64,
+    /// Number of compute units that a transaction or individual instruction is
+    /// allowed to consume. Compute units are consumed by program execution,
+    /// resources they use, etc...
+    pub compute_unit_limit: u64,
```

Comment:
Don't see the clarity this brings, `ComputeBudget::max_units` seems pretty clear, `ComputeBudget::compute_unit_limit` doesn't add much, at least not worth the churn it creates.

Reply:
I feel that `ComputeBudget::max_units` is confusingly named because it's only the max units for a particular tx, not the whole runtime. What churn are you concerned about? Just this PR itself?

Reply:
the budget is applied per tx.

Yeah, I was thinking about historic naming and doc churn.  Don't feel too strongly opposed to the changes.